src/iqstat.cpp

```diff
@@ -276,58 +144,64 @@ main( int argc, char **argv ) {
         usage();
         return 0;
     }
+    const auto disconnect = irods::at_scope_exit{[] { rcDisconnect(Conn); }};
 
-    status = getRodsEnv( &myEnv );
-    if ( status < 0 ) {
-        rodsLog( LOG_ERROR, "main: getRodsEnv error. status = %d",
-                 status );
-        return 1;
-    }
+    try {
+        status = getRodsEnv( &myEnv );
+        if ( status < 0 ) {
+            // TODO might have to change this for 4.3.0
 
-    if ( myRodsArgs.user ) {
-        snprintf( userName, sizeof( userName ), "%s", myRodsArgs.userString );
-    }
-    else {
-        snprintf( userName, sizeof( userName ), "%s", myEnv.rodsUserName );
-    }
+            rodsLog( LOG_ERROR, "main: getRodsEnv error. status = %d",
+                     status );
+            return 1;
+        }
 
-    // =-=-=-=-=-=-=-
-    // initialize pluggable api table
-    irods::api_entry_table&  api_tbl = irods::get_client_api_table();
-    irods::pack_entry_table& pk_tbl  = irods::get_pack_table();
-    init_api_table( api_tbl, pk_tbl );
+        if ( myRodsArgs.user && myRodsArgs.all ) {
+            std::fprintf(stderr, "Incompatible options: -u and -a cannot be used together\n");
+            return 2;
+        }
 
-    Conn = rcConnect( myEnv.rodsHost, myEnv.rodsPort, myEnv.rodsUserName,
-                      myEnv.rodsZone, 0, &errMsg );
+        if ( myRodsArgs.user ) {
+            snprintf( userName, sizeof( userName ), "%s", myRodsArgs.userString );
+        }
+        else {
+            snprintf( userName, sizeof( userName ), "%s", myEnv.rodsUserName );
+        }
 
-    if ( Conn == NULL ) {
-        return 2;
-    }
+        // =-=-=-=-=-=-=-
+        // initialize pluggable api table
+        irods::api_entry_table&  api_tbl = irods::get_client_api_table();
+        irods::pack_entry_table& pk_tbl  = irods::get_pack_table();
+        init_api_table( api_tbl, pk_tbl );
 
-    const auto disconnect = irods::at_scope_exit{[] { rcDisconnect(Conn); }};
+        Conn = rcConnect( myEnv.rodsHost, myEnv.rodsPort, myEnv.rodsUserName,
+                          myEnv.rodsZone, 0, &errMsg );
 
-    status = clientLogin( Conn );
-    if ( status != 0 ) {
-        if ( !debug ) {
-            return 3;
+        status = clientLogin( Conn );
+        if ( status != 0 ) {
+            if ( !debug ) {
+                return 3;
+            }
         }
-    }
 
-    nArgs = argc - myRodsArgs.optind;
-    if ( nArgs > 0 ) {
-        status = showRuleExec( userName, argv[myRodsArgs.optind],
-                               myRodsArgs.all );
-    }
-    else {
-        if ( myRodsArgs.longOption == 0 ) {
-            status = showRuleExecBrief( userName, myRodsArgs.all );
-        }
-        else {
-            status = showRuleExec( userName, "", myRodsArgs.all );
+        int nArgs { argc - myRodsArgs.optind };
```

Comment:
Unless I'm missing something, I think your `try` block can start right above this line here. Nothing above this throws as far as I know... Would reduce your diff a little bit, too. Up to you.

Reply:
Yes, doing that....